File: PythonIntro/CSharp/04_02AbstractClasses.cs
using System;

namespace CSharp
{
    public interface IAnimal
    {
        string GetAvgWeight();
    }

    public interface ITalker
    {
        string Talk();
    }

    public class Duck : IAnimal, ITalker
    {
        public string GetAvgWeight()
        {
            return "10kg";
        }

        public string Talk()
        {
            return "Quack";
        }

        public void Run()
        {
            var duck = new Duck();
            Console.WriteLine("Duck average weight: {0}", duck.GetAvgWeight());
            Console.WriteLine("Duck talks: {0}", duck.Talk());
        }
    }

}

File: PythonIntro/Python/04_02AbstractClasses.py
from abc import ABCMeta, abstractmethod

class Animal(object, metaclass=ABCMeta):

    @abstractmethod
    def getAvgWeight(self):
        pass

class Talker(object, metaclass=ABCMeta):

    @abstractmethod
    def talk(self):
        pass

class Duck(Animal, Talker): 
    def getAvgWeight(self):
        return "10kg"

    def talk(self):
        return "Quack"


if __name__ == "__main__":
    Animal.register(Duck)
    Talker.register(Duck)

    #animal = Animal()
    duck = Duck()

    print("Duck average weight: ", duck.getAvgWeight())
    print("Duck talks: ", duck.talk())